risc0/zkvm/src/serde/serializer.rs

```diff
@@ -120,11 +122,11 @@ impl<'a, W: WordWrite> serde::ser::Serializer for &'a mut Serializer<W> {
         false
     }
 
-    fn collect_str<T>(self, _: &T) -> Result<()>
+    fn collect_str<T>(self, value: &T) -> Result<()>
     where
-        T: core::fmt::Display + ?Sized,
+        T: ?Sized + core::fmt::Display,
     {
-        panic!("collect_str")
+        self.serialize_str(&value.to_string())
```

Comment:
- I see all the way back in https://github.com/risc0/risc0/pull/22 is where this panic was intentionally introduced. I don't know why this was done... perhaps there was a fundamental reason @flaub?

Reply:
This code was inspired by what the serde crate does. I'm not sure why it was implemented as a panic but in the past, I've fixed floating point numbers panic on serde so I think it was left out in the initial implementation. The tests are passing so we're definitely not regressing AFAIK